docs/blog/2022-07-20-bun-is-the-future.mdx

```diff
@@ -0,0 +1,239 @@
+
+This isn't the first time someone has tried to speed up NPM. You might be familiar with Yarn's Plug-n-Play and how it completely ditches the `node_modules` folder to try to get
+faster installs.  It does *work* but in practice,
+it's really hard to get working, and requires a lot of polyfilling and escape-hatches. We bit-the-bullet to use it, but I probably wouldn't recommend it or use it again.
+
+Bun works the way NPM does, just faster. It uses its own lockfile format, but node_modules and the package.json look normal.
+
+Now, Bun [doesn't have](https://github.com/oven-sh/bun/issues/533) workspace support yet, meaning it won't work for a lot of the big monorepos
+that could really benefit from it (ahem, us). That said, Bun development seems to be moving at a lightning pace. Workspace support is mentioned on the roadmap as of a couple of weeks ago.
+
+:::info
+Note that you don't have to switch entirely to Bun to use the package manager.  The same goes for the transpiler or the interpreter.
+You can choose just the part you need and leave the rest.  I imagine a _lot_ of people switching to this package manager in the near future without taking all of Bun.
+Perhaps some more people will learn about the project that way.
+:::
+
+## The Transpiler / Bundler
+
+Bun includes a transpiler for web browsers, to compete with webpack and esbuild. By the way, the parser in Bun is a Zig port
+of esbuild's parser. Neat.
+
+It already supports quite an impressive array of file types. Css, SVG, tsx, jsx, ts. They're all supported. Advanced options like CSS in JS appear to work fine.
+
+Since Bun includes a project scaffolder with a few built in templates, I simply called:
+
+```shell
+bun create react my-app
+```
+
+Then I ran `bun dev` and I had a react app running in my browser. I guess we can add `react-scripts` to the list of tools that Bun replaces.
+
+Changing the file extensions from `jsx` to `tsx` worked right away. Importing SVGs worked great. There also appears to be HMR support in dev mode,
+which has become a must-have for frontend devs coming from webpack.
+
+### What's missing from the transpiler?
+A lot of things you probably need for production, unfortunately. Minification is the biggest one I see still on the roadmap that real-world users will need.
+Having a large plugin ecosystem is kind of critical for a bundler to support different file formats. For instance, .vue files and .scss are still on the roadmap.
+Just about everyone and their grandma uses .scss, so that's a deal-breaker.
+
+## Other features - Web server and sqlite client
+
+Bun also adds some things you'd usually think of as frameworks to the standard library. Personally, I'm less excited about these more library type features. There are a lot of very good options available in Node for http servers.
+
+Bun's webserver looks pretty bare-bones. Express still works fine for most of the community, even if it is a bit behind the times. (They just added promise support this year).
```

Comment:
you could mention it uses the sendfile syscall so it's more performant on linux as a counterpoint to it being bare bones
https://twitter.com/jarredsumner/status/1506182333459693571